src/main/java/Thames/Parser.java

```diff
@@ -0,0 +1,85 @@
+package Thames;
+
+import Thames.command.;
+import Thames.task.ToDo;
+import Thames.task.Deadline;
+import Thames.task.Event;
+
+/**
+ * Class that makes sense of the user input.
+ */
+public class Parser {
+    /**
+     * Makes sense of the user input to return Command object.
+     *
+     * @param fullCommand User input.
+     * @return Command that contains the properties of user input.
+     * @throws ThamesException If user input is invalid.
+     */
+    static Command parse(String fullCommand) throws ThamesException {
+        String[] split = fullCommand.trim().split(" ", 2);
+        String input = split[0];
+
+        switch(input.toLowerCase()) {
+        case "todo":
+            if (split.length != 2) {
+                throw new ThamesException("Please provide your Todo task in the following format: \n" +
```

Comment:
When wrapping around, ensure that symbols such as '+', ':', etc. are at the start of the following line. This issue is quite consistent throughout your code. Maybe consider using the Gradle checkstyle plugin.